### PR TITLE
feat: add pnpm support

### LINKED ISSUE #145: Fails if using pnpm
We are using pnpm (https://pnpm.js.org/) with our project. As such, we do not have a `package-lock.json` file. Even setting install to false fails with the cypress github action (see below).

Pnpm uses a `pnpm-lock.yaml` file that serves the same purpose, and generally (though not always) has the same syntax as npm. However, the lockfile formats are not compatible.

>This action installs local dependencies using lock files. If yarn.lock file is found, the install uses yarn --frozen-lockfile command. Otherwise it expects to find package-lock.json and install using npm ci command.

```
internal/fs/utils.js:220
    throw err;
    ^

Error: ENOENT: no such file or directory, open 'package-lock.json'
    at Object.openSync (fs.js:440:3)
    at Object.readFileSync (fs.js:342:35)
    at Function.module.exports.309.hasha.fromFileSync (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:4439:54)
    at Object.104 (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:2440:24)
    at __webpack_require__ (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:22:30)
    at startup (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:37:19)
    at /home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:41:18
    at Object.<anonymous> (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:44:10)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'package-lock.json'
}
```

If relevant, there is a github action for pnpm: https://github.com/marketplace/actions/setup-pnpm
They also have these docs that might be relevant: https://pnpm.js.org/en/continuous-integration

### PR DESCRIPTION
**Pull request by penx**
This is branched @FFdhorkin's work at https://github.com/FFdhorkin/github-action/tree/feature/pnpm-support but currently still has a few issues.

Fixes #145

### CODE REVIEW

**Comment by lifeiscontent** [comment]
@jennifer-shehane any idea when the cypress team might take a look at this PR? Really would love to use these improvements.

**Comment by FFdhorkin** [comment]
@penx regarding the error you are getting:
>Unable to process command '::add-path::/home/runner/setup-pnpm/.bin' successfully.
>
>The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`.

Perhaps this is due to the older version of PNPM's github action in this PR? The example file uses `pnpm/action-setup@v1.2.0` but the latest is `pnpm/action-setup@v2.0.1` (not to mention pnpm 5.4.3 is superceded by 6.3.x now)

File: examples/basic-pnpm/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "index.js",
+  "scripts": {
+    "preinstall": "npx only-allow pnpm",
+    "test": "cypress run"
+  },
+  "author": "",
+  "license": "ISC",
+  "private": true,
+  "devDependencies": {
+    "cypress": "4.8.0"
```
Comment (ebeloded) [inline examples/basic-pnpm/package.json:14]: The current version of Cypress is 7.5

**Comment by ebeloded** [comment]
How can use this version of the action?

**Comment by penx** [comment]
Hi all, just to let you know I'm taking a break for a couple weeks and will pick this up again after that. If someone wants to pick it up in the meantime they are more than welcome to.

Also check #382 which is what I personally need and is very similar.

**Comment by lifeiscontent** [comment]
@bahmutov can you take a look at this PR?

**Comment by penx** [comment]
I've:
- updated the following dependencies to the latest versions
  - cypress
  - pnpm/action-setup
  - pnpm
- Rebased off master and rebuilt dist
- Updated the pnpm lock file

but this still fails with

```
Run ./
  with:
    working-directory: examples/basic-pnpm
    build: pnpx cypress info
    record: false
    config-file: cypress.json
/home/runner/setup-pnpm/node_modules/.bin/pnpm install --frozen-lockfile -C
undefined
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command failed with ENOENT: install --frozen-lockfile -C
spawn install --frozen-lockfile -C ENOENT
Error: The process '/home/runner/setup-pnpm/node_modules/.bin/pnpm' failed with exit code 1
```
https://github.com/penx/github-action/runs/3589281020?check_suite_focus=true

**Comment by penx** [comment]
This was intentionally in draft as I was picking up someone else's PR to see if it would build on CI with some minor modifications.

It didn't, and I'm not sure what to do next to resolve the issues I'm afraid. I don't personally need pnpm support but will be happy to try out any suggestions - or even better if someone wants to pick it up then feel free.

**Comment by penx** [comment]
I think the comment here https://github.com/cypress-io/github-action/issues/145#issuecomment-884808726 may also be the way forward - i.e. if you are using pnpm, manage the dependency install yourself and then pass `install:false` to this action.

**Comment by penx** [comment]
Closing in favour of the workaround detailed above

**Pull request closed without merging**